pkg/projectfile/projectfile.go

```diff
@@ -464,6 +466,18 @@ func (p *Project) Init() error {
 	}
 	p.parsedURL = parsedURL
 
+	if p.parsedURL.CommitID != "" {
+		// Note: do not migrate to localcommit here. Writing to disk during Parse() feels wrong.
+		// Migrate when SetCommit() is called instead.
+		p.commitID = p.parsedURL.CommitID
+	} else {
+		if commitID, err := localcommit.Get(filepath.Dir(p.Path())); err == nil {
+			p.commitID = commitID
```

Comment:
As noted in my other comment; this is no longer the responsibility of projectfile.

Any code that is currently accessing `projectfile.CommitID()` will have to be transitioned to instead use `localcommit.Get()`. For what it's worth in Goland you can use refactoring tools to make this change in a matter of seconds.